File: code-kata/Absolute-beginner/Leap_year_or_not.py
# Let "A" be a year, write a program to check whether this year is a leap year or not.

year=int(input())
if (year % 4) == 0:
    if (year % 100) == 0:
        if (year % 400) == 0:
           print("Y",end="")
        else:
           print("N",end="")
    else:
       print("Y",end="")
else:
    print("N",end="")

File: javascript/codekata/Leap_year_or_not.js
// Print "Y" if its a leap year and "N" if its a common year.

const readline=require('readline');
const inp=readline.createInterface({
  input:process.stdin});
const n=[];
inp.on("line",(data)=>{
  n.push(data);});
inp.on("close",()=>{
  var year=parseInt(n[0]);
  if((year%4)==0){
    if((year%100)==0){
      if((year%400)==0){
        console.log("Y")}
      else{
        console.log("N")}}
    else{
      console.log("Y")}}
  else{
    console.log("N")}

});
